src/wallet/wallet.ts

```diff
@@ -1352,7 +1354,14 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
 
     if (newOptions.createAnotherMint) {
       // b. Mint authority
-      outputsObj.push(new Output(TOKEN_MINT_MASK, p2pkhScript, {tokenData: AUTHORITY_TOKEN_DATA}));
+      const authorityAddress = newOptions.mintAuthorityAddress || this.getCurrentAddress({ markAsUsed: true }).address;
+      const authorityAddressObj = new Address(authorityAddress, { network: this.network });
+      if (!authorityAddressObj.isValid()) {
+        throw new SendTxError(`Address ${newOptions.mintAuthorityAddress} is not valid.`);
+      }
+      const p2pkhAuthority = new P2PKH(authorityAddressObj);
```

Comment:
Should we create a method `Address.getP2PKHScript()`? It seems the same pattern has been repeating in this part of the code.

Reply:
Done!